src/client/datascience/jupyter/jupyterUtils.ts

```diff
@@ -96,5 +96,5 @@ export async function computeWorkingDirectory(resource: Resource, workspace: IWo
     }
 
     // Otherwise a file without an extension or directory doesn't exist. Just use the workspace root
-    return workspace.getWorkspaceFolder(resource)?.uri.fsPath || process.cwd();
+    return workspace.getWorkspaceFolder(resource)?.uri.fsPath || workspace.rootPath || process.cwd();
```

Comment:
Seems getWorkspaceFolder only works if you're in a workspace.

Reply:
FYI - Under the hood `workspace.rootPath` also uses `workspace.worksapceFolders`, hence if you don't have a workspace folder opened, then `workspace.root` will still return `undefined`.

Reply:
No change required.